src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -126,120 +128,124 @@ inline data_types element_type_to_data_type(ov::element::Type t) {
-        return _dynamic_pad_dims;
-    }
     /// @brief
-    /// @param lower_sizes Top-left padding sizes. See @ref tensor::tensor(const std::vector<value_type>&, value_type) for details.
-    /// @param upper_sizes Bottom-right padding sizes. See @ref tensor::tensor(const std::vector<value_type>&, value_type) for details.
-    /// @param filling_value Filling value for padding area.
-    padding(const std::vector<tensor::value_type>& lower_sizes,
-            const std::vector<tensor::value_type>& upper_sizes,
-            float filling_value = 0.0f,
-            const tensor& dynamic_pad_dims = tensor(0))
-        : _lower_size(to_abs(lower_sizes), 0),
-          _upper_size(to_abs(upper_sizes), 0),
-          _filling_value(filling_value),
-          _dynamic_pad_dims(dynamic_pad_dims) {}
+    /// @param lower_sizes Top-left padding sizes, in the same size and order as shape.
+    /// @param upper_sizes Bottom-right padding sizes, in the same size and order as shape.
+    padding(const std::vector<int32_t>& lower_sizes,
+            const std::vector<int32_t>& upper_sizes,
+            const DynPadDimsMask& dynamic_pad_dims = EMPTY_MASK) {
+            // paddings
+            OPENVINO_ASSERT(lower_sizes.size() <= SHAPE_RANK_MAX);
+            OPENVINO_ASSERT(upper_sizes.size() <= SHAPE_RANK_MAX);
+            if (lower_sizes.size() > 0) std::copy_n(lower_sizes.begin(), lower_sizes.size(), _lower_size.begin());
+            if (upper_sizes.size() > 0) std::copy_n(upper_sizes.begin(), upper_sizes.size(), _upper_size.begin());
+            _dynamic_pad_dims = dynamic_pad_dims;
+          }
 
     /// @brief Constrcuts symmetric padding.
-    /// @param sizes Top-left and bottom-right padding sizes. See @ref tensor::tensor(const std::vector<value_type>&,
-    /// value_type) for details.
-    /// @param filling_value Filling value for padding area.
-    explicit padding(const std::vector<tensor::value_type>& sizes, float filling_value = 0.0f, const tensor& dynamic_pad_dims = tensor(0))
-        : padding(sizes, sizes, filling_value, dynamic_pad_dims) {}
+    /// @param sizes Top-left and bottom-right padding sizes, in the same size and order as shape.
+    explicit padding(const std::vector<int32_t>& sizes,
+                     const DynPadDimsMask& dynamic_pad_dims = EMPTY_MASK)
+        : padding(sizes, sizes, dynamic_pad_dims) {}
 
     /// @brief Constructs "zero-sized" padding.
-    padding() : padding({0, 0, 0, 0}, 0, tensor(0)) {}
+    padding() : padding({}, EMPTY_MASK) {}
 
     /// @brief Returns true if padding size is not zero.
     explicit operator bool() const {
-        return std::any_of(_lower_size.raw.begin(), _lower_size.raw.end(), [](const tensor::value_type& el) { return el != 0; }) ||
-               std::any_of(_upper_size.raw.begin(), _upper_size.raw.end(), [](const tensor::value_type& el) { return el != 0; });
+        return std::any_of(_lower_size.begin(), _lower_size.end(), [](int i){ return i > 0; }) ||
```

Comment:
Maybe we should check mask here as well?

Reply:
For build time, maybe. But this method could also be called during runtime. Unluckily padding class has no info of runtime/build-time.

On the other hand, however, check mask here (and return true for dynamic mask) will result in failure of test case prepare_buffer_fusing.in_place_crop_dynamic_reshape_unsqueeze. 
1. Bellow action will set dynamic dims mask for reshape node, which is originally not padded.
> https://github.com/sshlyapn/openvino/blob/dceafd8a9fbed8a554fdab99753c85e1c08dbe89/src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp#L688

2. If mask is checked and return true, reshape node will be not optimized due to bellow check -

> https://github.com/sshlyapn/openvino/blob/dceafd8a9fbed8a554fdab99753c85e1c08dbe89/src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp#L842

In this case, the reshape node is expected to be optimized. Probably we should re-consider this code logic here - why do we set dyn dims mask here?

Reply:
OK, let's keep it as is for now